src/TextInput/index.js

```diff
@@ -79,20 +84,43 @@ class TextInput extends React.Component<Props, State> {
     }));
   };
 
-  onFocus = () => {
+  handleOnFocus = () => {
     const { onFocus, disabled } = this.props;
-    if (!disabled) {
-      this.toggleFocus();
-      onFocus && onFocus();
+    if (disabled) {
+      return;
     }
+
+    this.toggleFocus();
+    onFocus && onFocus();
   };
 
-  onBlur = () => {
+  handleOnBlur = () => {
     const { onBlur, disabled } = this.props;
-    if (!disabled) {
-      this.toggleFocus();
-      onBlur && onBlur();
+    if (disabled) {
+      return;
     }
+
+    this.toggleFocus();
+    onBlur && onBlur();
+  };
+
+  handleChangeText = (value: string) => {
+    const { onChangeText, disabled } = this.props;
+    if (disabled) {
+      return;
+    }
+
+    onChangeText && onChangeText(value);
```

Comment:
Wow, this is weird, shouldn't it be:

```js
if (!disabled && onChangeText) { 
  onChangeText(value);
}

```

On the other hand, when is disabled (`editable = false`) are you sure this function is not already not called?